src/core/toga/sources/dicts.py

```diff
@@ -0,0 +1,73 @@
+from .base import Source, Node, Datum, to_accessor
+
+
+class BaseDictSource(Source):
```

Comment:
Why are we defining parent classes? What expansion are we expecting?

Reply:
Also you have a nice comment describing what `ListSource` is for, it would be nice to see something similar for `DictSource`

Reply:
We're expecting end users to define their own data sources.

Consider the use case of Cricket. Cricket has, at it's core, a data source - a representation of the test suite. That test suite can be visualised as a tree, but the insert/remove style API of the DictDataSource isn't helpful - you want it to be automatically updated as a result of loading a new test suite, or a new test being found.

Sure - we could have 2 representations - an underlying collection of test data, and a DictSource - but the DictSource is just an API for retrieving data in a known format, so why not allow the underlying test data to reflect that API?

The base Class is there to differentiate between primitive operations that *must* be implemented (like insert), and composite operations that can be re-used once the primitives are available.